indexer/packages/postgres/src/stores/trading-reward-table.ts

```diff
@@ -0,0 +1,105 @@
+import { QueryBuilder } from 'objection';
+
+import { BUFFER_ENCODING_UTF_8, DEFAULT_POSTGRES_OPTIONS } from '../constants';
+import { setupBaseQuery, verifyAllRequiredFields } from '../helpers/stores-helpers';
+import Transaction from '../helpers/transaction';
+import { getUuid } from '../helpers/uuid';
+import TradingRewardModel from '../models/trading-reward-model';
+import {
+  Options,
+  Ordering,
+  QueryableField,
+  QueryConfig,
+  TradingRewardColumns,
+  TradingRewardCreateObject,
+  TradingRewardFromDatabase,
+  TradingRewardQueryConfig,
+} from '../types';
+
+export function uuid(address: string, blockHeight: string): string {
+  // TODO(IND-483): Fix all uuid string substitutions to use Array.join.
+  return getUuid(Buffer.from(`${address}-${blockHeight}`, BUFFER_ENCODING_UTF_8));
+}
+
+export async function findAll(
+  {
+    address,
+    blockHeight,
+    blockTimeBeforeOrAt,
+    limit,
+  }: TradingRewardQueryConfig,
+  requiredFields: QueryableField[],
+  options: Options = DEFAULT_POSTGRES_OPTIONS,
+): Promise<TradingRewardFromDatabase[]> {
+  verifyAllRequiredFields(
+    {
+      address,
+      blockHeight,
+      blockTimeBeforeOrAt,
+      limit,
+    } as QueryConfig,
+    requiredFields,
+  );
+
+  let baseQuery: QueryBuilder<TradingRewardModel> = setupBaseQuery<TradingRewardModel>(
+    TradingRewardModel,
+    options,
+  );
+
+  if (address) {
+    baseQuery = baseQuery.where(TradingRewardColumns.address, address);
+  }
+
+  if (blockHeight) {
+    baseQuery = baseQuery.where(TradingRewardColumns.blockHeight, blockHeight);
+  }
+
+  if (blockTimeBeforeOrAt) {
+    baseQuery = baseQuery.where(TradingRewardColumns.blockTime, '<=', blockTimeBeforeOrAt);
+  }
+
+  if (options.orderBy !== undefined) {
+    for (const [column, order] of options.orderBy) {
+      baseQuery = baseQuery.orderBy(
+        column,
+        order,
+      );
+    }
+  } else {
+    baseQuery = baseQuery.orderBy(
+      TradingRewardColumns.blockHeight,
+      Ordering.ASC,
+    );
+  }
```

Comment:
nit: order by block time and address by default for a deterministic ordering.

Reply:
I did block height and address which should be the same